metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/telemetry/MonitorFactory.java

```diff
@@ -86,15 +94,72 @@ public Monitor createOrUpdateMonitor(final String id) {
 		return createOrUpdateMonitor(attributes, resource, monitorType, id);
 	}
 
+	/**
+	 * Retrieves monitor job keys from the telemetry manager's connector store.
+	 *
+	 * <p>Returns the keys from a monitor job associated with the given `connectorId`
+	 * and `monitorType`.
+	 * Logs an error and returns null if the connector store or monitor job is missing.
+	 *
+	 * @return a set of monitor job keys, or null if unavailable.
+	 */
+	private LinkedHashSet<String> getMonitorJobKeys() {
```

Comment:
Would it be possible please to remove this method and build the `MonitorFactory `with the keys directly from the strategy that processes the monitors (same type monitors).